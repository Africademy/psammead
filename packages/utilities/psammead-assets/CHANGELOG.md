<h1 align="center">Psammead Assets Changelog</h1>

| Version | Description |
|---------|-------------|
| 0.1.0   | Create initial package, pulled in from [simorgh](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md). [PR #49](https://github.com/BBC-News/psammead/pull/49) |
