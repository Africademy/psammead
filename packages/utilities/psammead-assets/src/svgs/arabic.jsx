import React from 'react';

const arabic = {
  group: (
    <g fill="none" fillRule="evenodd">
      <path
        fill="#FFF"
        d="M72.882 15.678l-.372.227c-1.798 1.098-3.754 1.687-5.53 1.686-3.802-.014-6.306-2.267-6.317-5.548.005-3.207 2.621-5.581 6.221-5.59 1.834.002 3.495.476 5.469 1.557l.362.2V5.281l-.154-.06c-2.227-.88-3.95-1.271-5.652-1.27-2.633-.001-4.917.868-6.544 2.344-1.627 1.475-2.594 3.558-2.592 5.95.001 1.667.688 3.62 2.16 5.17 1.469 1.55 3.73 2.681 6.833 2.679h.01c2.573 0 4.357-.574 5.969-1.367l.137-.067v-2.982zm-18.606 7.261h23.219V1.089H54.276v21.85zM45.223 15.576c.007-1.986-1.315-3.494-3.431-4.074.571-.292 1.023-.649 1.353-1.084.443-.583.654-1.298.652-2.13a3.801 3.801 0 0 0-1.265-2.849c-.866-.78-2.179-1.287-3.97-1.287h-4.949v15.737h5.808c2.073 0 3.527-.561 4.464-1.393a3.882 3.882 0 0 0 1.338-2.92zm-17.702 7.363H50.74V1.089H27.521v21.85z"
      />
      <path
        fill="#FFF"
        d="M42.334 15.262c-.003.52-.162 1.043-.628 1.451-.463.409-1.267.719-2.597.719H36.43v-4.295h2.56c1.193 0 2.04.241 2.571.619.529.381.768.88.773 1.506M40.165 10.119c.497-.361.739-.868.742-1.63-.002-.524-.152-.972-.529-1.306-.378-.335-1.018-.573-2.053-.573H36.43v4.07h1.301c1.125.001 1.931-.2 2.434-.561M18.47 15.576c.007-1.986-1.316-3.494-3.431-4.074.571-.292 1.021-.649 1.352-1.084.443-.583.654-1.298.652-2.13a3.8 3.8 0 0 0-1.264-2.849c-.866-.78-2.179-1.287-3.971-1.287H6.859v15.737h5.808c2.074 0 3.527-.561 4.463-1.393.937-.83 1.341-1.917 1.34-2.92zM.768 22.939h23.219V1.089H.768v21.85z"
      />
      <path
        fill="#FFF"
        d="M13.41 10.119c.498-.361.74-.868.744-1.63-.001-.524-.152-.972-.53-1.306-.378-.335-1.018-.573-2.053-.573H9.678v4.07h1.299c1.125.001 1.932-.2 2.433-.561M14.807 13.756c-.53-.378-1.378-.619-2.57-.619H9.678v4.295h2.678c1.33 0 2.132-.31 2.597-.719.465-.408.625-.931.627-1.451-.004-.626-.243-1.125-.773-1.506M101.81 4.149h2.146v15.785h-1.943L91.46 7.777v12.157h-2.125V4.149h1.831l10.644 12.26V4.149M108.029 4.149h8.949V6.16h-6.69v4.814h6.463v2.023h-6.463V17.9h6.904v2.011h-9.163V4.149M141.094 4.149h2.253l-6.381 15.852h-.496l-5.161-12.835-5.215 12.835h-.485l-6.362-15.852h2.277l4.35 10.881 4.378-10.881h2.137l4.399 10.881 4.306-10.881M149.471 13.32l-1.718-1.043c-1.077-.658-1.843-1.305-2.299-1.942-.456-.636-.683-1.369-.683-2.197 0-1.244.431-2.253 1.294-3.028.862-.776 1.982-1.164 3.36-1.164 1.319 0 2.528.37 3.627 1.11v2.565c-1.137-1.094-2.361-1.642-3.672-1.642-.738 0-1.343.171-1.819.513-.475.341-.712.778-.712 1.311 0 .472.175.915.524 1.328.35.413.913.846 1.687 1.3l1.727 1.021c1.924 1.15 2.887 2.614 2.887 4.391 0 1.266-.424 2.294-1.271 3.085-.848.791-1.949 1.186-3.305 1.186-1.559 0-2.979-.479-4.26-1.438v-2.869c1.22 1.546 2.633 2.318 4.238 2.318.708 0 1.297-.196 1.767-.59.471-.394.707-.888.707-1.481 0-.961-.693-1.872-2.079-2.734"
      />
      <path stroke="#FFF" strokeWidth="2" d="M169.143.813v22.594" />
      <path
        fill="#FFF"
        d="M203.795 5.452l-.001-.003a6.548 6.548 0 0 0-.323-.583c-.006.01-.024.023-.057.047l-1.69 1.461c-.019.02-.019.034-.019.045l.036.063c.035.043.08.106.135.187l.173.257a.76.76 0 0 0 .102.144c.727 1.031 1.053 2.149 1.08 3.659l.005.254h-10.399l.417 1.76c.025.135.036.196.3.196h2.864l-.156.351c-.789 1.77-2.261 2.705-4.258 2.705-2.489 0-4.517-1.977-5.437-5.295l-.795 1.717c.057.557.568 1.829 1.069 2.639 1.228 1.927 3.302 2.896 5.603 2.896 2.726 0 5.445-2.162 6.061-4.82l.045-.193h6.429c.111-1.073.156-1.527.156-1.921 0-.199 0-.35-.024-.624-.128-2.112-.459-3.358-1.316-4.942M201.117 16.046a.282.282 0 0 0-.056.161c0 .04.026.104.055.132l1.007 1.009a.277.277 0 0 0 .162.056c.044 0 .067-.009.16-.079l.982-.985c.051-.052.08-.101.08-.133 0-.04-.007-.085-.082-.161l-1.008-1.007a.177.177 0 0 0-.132-.056.26.26 0 0 0-.162.056l-1.006 1.007M190.75 19.173a.243.243 0 0 0-.159.057l-1.007 1.031a.184.184 0 0 0-.057.136c0 .049.013.099.079.187l.959.955c.048.048.056.056.16.056a.178.178 0 0 0 .133-.056l1.036-1.01a.233.233 0 0 0 .055-.132.315.315 0 0 0-.058-.16l-.993-.994c-.087-.062-.132-.07-.148-.07"
      />
      <path
        fill="#FFF"
        d="M193.172 19.173c-.048 0-.099.014-.187.08l-.978 1.008c-.058.057-.058.07-.058.136 0 .049.013.099.078.186l.982.954a.272.272 0 0 0 .163.058c.088 0 .1-.014.111-.03l1.03-1.034c.053-.052.083-.101.083-.134a.255.255 0 0 0-.083-.159l-1.008-1.008a.175.175 0 0 0-.133-.057M219.615 11.009c-1.518 0-2.497-.93-2.497-2.368 0-1.281.678-2.337 3.917-2.337.158 0 .316 0 .552.011V5.123a13.104 13.104 0 0 0-2.49-.238c-2.432 0-4.195 1.449-4.195 3.446 0 .896.198 1.538.707 2.289l.262.389h-4.519l-.027-.221a16.373 16.373 0 0 0-.84-3.616c-.14-.398-.337-.966-.45-1.196l-1.697 1.55a.197.197 0 0 0-.006.056c0 .072.037.163.123.315.749 1.31 1.144 2.836 1.144 4.414 0 .999-.153 1.663-.549 2.369-.607.995-1.702 1.608-3.257 1.83l-.046.006 1.49 1.42c2.522-.402 4.042-2.237 4.203-5.05.089.046.203.08.369.08l10.642-.001c.211 0 .32-.055.32-.165v-1.791h-3.156"
      />
    </g>
  ),
  ratio: 9.2917,
  viewbox: {
    height: 24,
    width: 223,
  },
};

export default arabic;
