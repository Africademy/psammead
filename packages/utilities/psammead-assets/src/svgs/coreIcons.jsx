import React from 'react';
import styled from 'styled-components';
import { C_EBON } from '@bbc/psammead-styles/colours';
import {
  GEL_SPACING_HLF,
  GEL_SPACING_DBL,
} from '@bbc/gel-foundations/spacings';

// `currentColor` has been used to better reflect user colour choices in Firefox.
const CoreIcon = styled.svg`
  vertical-align: middle;
  margin: 0 ${GEL_SPACING_HLF};
  color: ${C_EBON};
  fill: currentColor;
`;

const AlertCoreIcon = styled(CoreIcon)`
  height: ${GEL_SPACING_DBL};
  width: ${GEL_SPACING_DBL};
`;

const InfoCoreIcon = styled(CoreIcon)`
  width: ${GEL_SPACING_DBL};
  height: ${GEL_SPACING_DBL};
`;

const CoreIcons = {
  alert: (
    <AlertCoreIcon
      viewBox="0 0 32 32"
      width="32px"
      height="32px"
      focusable="false"
    >
      <path d="M16 2L0 30h32zm2 25h-4v-4h4zm-4-6V11h4v10z" />
    </AlertCoreIcon>
  ),
  info: (
    <InfoCoreIcon
      viewBox="0 0 32 32"
      width="32px"
      height="32px"
      focusable="false"
    >
      <path d="M16 0a16 16 0 1 0 16 16A16 16 0 0 0 16 0zm2 25h-4V13h4zm0-14h-4V7h4z" />
    </InfoCoreIcon>
  ),
};

export default CoreIcons;
