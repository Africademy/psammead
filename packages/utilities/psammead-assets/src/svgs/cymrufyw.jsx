import React from 'react';

const cymrufyw = {
  group: (
    <g
      fillRule="evenodd"
      stroke="#000"
      strokeWidth=".335"
      style={{ stroke: '#fff' }}
    >
      <path d="M39.396 24h1.359V0h-1.359zM49.938 9.687c-.333-.153-.694-.265-1.263-.265-1.154 0-1.89.924-1.89 2.28 0 1.33.625 2.352 1.93 2.352.53 0 1.112-.197 1.434-.337l.082 1.553c-.542.167-1.098.238-1.863.238-2.347 0-3.528-1.609-3.528-3.806 0-2.085 1.223-3.735 3.46-3.735.75 0 1.29.084 1.805.238zm4.78 3.708h.027l1.723-5.26h1.891l-2.683 7.149c-.597 1.595-1.029 3.161-2.973 3.161a4.72 4.72 0 01-1.32-.195l.124-1.385c.237.084.502.126.961.126.749 0 1.209-.517 1.209-1.287l-2.74-7.569h2.058zm4.724-5.26h1.736v1.007h.028c.584-.881 1.431-1.175 2.127-1.175.999 0 1.708.378 2.11 1.273.431-.867 1.363-1.273 2.237-1.273 1.765 0 2.264 1.218 2.264 2.826v4.547h-1.86v-4.294c0-.673 0-1.624-1.071-1.624-1.236 0-1.388 1.483-1.388 2.42v3.498h-1.862v-4.294c0-.673 0-1.624-1.07-1.624-1.237 0-1.39 1.483-1.39 2.42v3.498h-1.861zm12.007 0h1.654v1.637h.027c.084-.672.847-1.805 1.96-1.805.18 0 .373 0 .57.056v1.888c-.169-.098-.502-.153-.835-.153-1.515 0-1.515 1.903-1.515 2.937v2.645h-1.86zm11.99 7.205h-1.763v-.979h-.029c-.57.7-1.376 1.147-2.348 1.147-1.695 0-2.43-1.203-2.43-2.826V8.135h1.862v3.847c0 .882.012 2.071 1.208 2.071 1.348 0 1.639-1.468 1.639-2.392V8.135h1.861zM20.07 13.463c.003-.903-.596-1.59-1.55-1.854a1.77 1.77 0 00.611-.491c.201-.265.295-.59.295-.969 0-.466-.182-.94-.571-1.294-.393-.355-.985-.585-1.793-.585h-2.236v7.153h2.626c.936 0 1.592-.256 2.014-.632.42-.378.604-.873.604-1.328zm-7.994 3.346H22.56V6.876H12.076z" />
      <path d="M18.766 13.318a.842.842 0 01-.286.661c-.209.186-.57.327-1.17.327H16.1v-1.953h1.157c.537 0 .917.11 1.159.282.24.173.346.4.35.683m-.98-2.337c.224-.163.334-.393.334-.74 0-.238-.068-.442-.238-.593-.168-.153-.46-.261-.929-.261H16.1v1.85h.587c.508 0 .872-.092 1.098-.256m-9.793 2.482c.003-.903-.593-1.59-1.55-1.854.26-.132.463-.294.61-.491.201-.265.294-.59.294-.969 0-.466-.18-.94-.57-1.294-.39-.355-.981-.585-1.79-.585H2.751v7.153h2.622c.936 0 1.592-.256 2.013-.632a1.77 1.77 0 00.606-1.328zM0 16.809h10.481V6.876H.001z" />
      <path d="M5.708 10.981c.225-.163.334-.393.336-.74 0-.238-.07-.442-.238-.593-.172-.153-.462-.261-.928-.261h-.854v1.85h.587c.506 0 .872-.092 1.097-.256m.63 1.654c-.239-.172-.622-.282-1.16-.282H4.024v1.953H5.23c.602 0 .965-.14 1.175-.327a.85.85 0 00.282-.661.796.796 0 00-.35-.683m26.217.873l-.169.104c-.807.498-1.692.766-2.493.766-1.717-.006-2.85-1.03-2.855-2.522.005-1.459 1.187-2.537 2.808-2.541.832 0 1.582.217 2.471.708l.163.09v-1.33l-.07-.027c-1.007-.4-1.785-.579-2.553-.579-1.185 0-2.22.395-2.953 1.066-.732.672-1.169 1.618-1.169 2.706 0 .757.309 1.646.977 2.35.663.704 1.682 1.217 3.082 1.217h.007c1.161 0 1.964-.26 2.69-.622l.064-.029zm-8.401 3.301h10.483V6.876H24.153zM88.14 8.912h-1.443V8.15h1.443v-.52c0-1.51.056-3.033 1.89-3.033.28 0 .66.043.827.141l-.056.776a1.556 1.556 0 00-.7-.155c-1.205 0-1.078 1.311-1.078 2.172v.62h1.61v.762h-1.61v6.432h-.883zm6.61 7.503c-.337.989-.84 1.976-1.836 1.976a4.22 4.22 0 01-.952-.113l.084-.763c.168.057.336.113.658.113.757 0 1.009-.634 1.289-1.566l.182-.634L91.64 8.15h.966l2.003 6.008-.014-.012h.028l2.059-5.996h.925zm12.168-1.071h-1.064l-1.947-6.22-1.975 6.22h-1.064L98.57 8.15h.967l1.877 6.22 2.017-6.22h1.064l1.891 6.22 2.003-6.22h.882z" />
    </g>
  ),
  ratio: 4.5833,
  viewbox: {
    height: 24,
    width: 110,
  },
};

export default cymrufyw;
