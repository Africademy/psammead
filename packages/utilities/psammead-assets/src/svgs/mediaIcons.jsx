import React from 'react';
import styled from 'styled-components';
import { C_EBON } from '@bbc/psammead-styles/colours';
import {
  GEL_SPACING_HLF,
  GEL_SPACING_DBL,
} from '@bbc/gel-foundations/spacings';

// `currentColor` has been used to better reflect user colour choices in Firefox.
const MediaIcon = styled.svg`
  vertical-align: middle;
  margin: 0 ${GEL_SPACING_HLF};
  color: ${C_EBON};
  fill: currentColor;
`;

const VideoMediaIcon = styled(MediaIcon)`
  width: 0.75rem;
  height: 0.75rem;
`;

const AudioMediaIcon = styled(MediaIcon)`
  width: 0.8125rem;
  height: 0.75rem;
`;

const PhotoMediaIcon = styled(MediaIcon)`
  width: 1rem;
  height: 0.8125rem;
`;

const GuidanceIcon = styled(MediaIcon)`
  width: ${GEL_SPACING_DBL};
  height: ${GEL_SPACING_DBL};
`;

const mediaIcons = {
  video: (
    <VideoMediaIcon
      viewBox="0 0 32 32"
      width="12px"
      height="12px"
      focusable="false"
    >
      <polygon points="3,32 29,16 3,0" />
    </VideoMediaIcon>
  ),
  audio: (
    <AudioMediaIcon
      viewBox="0 0 13 12"
      width="13px"
      height="12px"
      focusable="false"
    >
      <path d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z" />
      <path d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z" />
    </AudioMediaIcon>
  ),
  photogallery: (
    <PhotoMediaIcon
      viewBox="0 0 32 26"
      width="16px"
      height="13px"
      focusable="false"
    >
      <path d="M9,2V0H4V2H0V26H32V2ZM6.5,10A2.5,2.5,0,1,1,9,7.52,2.5,2.5,0,0,1,6.5,10ZM20,23a9,9,0,1,1,9-9A9,9,0,0,1,20,23Z" />
      <circle cx="20" cy="14.02" r="5.5" />
    </PhotoMediaIcon>
  ),
  guidance: (
    <GuidanceIcon
      viewBox="0 0 24 24"
      focusable="false"
      height="24px"
      width="24px"
    >
      <g fill="none" fillRule="evenodd">
        <path
          fill="#fefefe"
          d="M24,12 C24,18.627 18.627,24 12,24 C5.3715,24 0,18.627 0,12 C0,5.373 5.3715,0 12,0 C18.627,0 24,5.373 24,12"
        />
        <path
          fill="#3F3F42"
          d="M17.515275,17.99265 L15.500775,17.99265 L15.316275,16.6839 C14.493525,17.8419 13.084275,18.24465 11.975025,18.24465 C8.349525,18.24465 5.999775,15.4749 5.999775,12.00015 C5.999775,8.5254 8.349525,5.75565 11.975025,5.75565 C14.459775,5.75565 17.112525,7.08165 17.398275,10.08615 L14.275275,10.08615 C14.174025,9.06315 13.167525,8.44215 11.907525,8.44215 C9.893775,8.44215 9.122025,10.2204 9.122025,12.00015 C9.122025,13.7799 9.893775,15.55965 11.907525,15.55965 C13.453275,15.55965 14.493525,14.8374 14.661525,13.74615 L12.479025,13.74615 L12.479025,11.4294 L17.515275,11.4294 L17.515275,17.99265 Z"
        />
      </g>
    </GuidanceIcon>
  ),
};

export default mediaIcons;
