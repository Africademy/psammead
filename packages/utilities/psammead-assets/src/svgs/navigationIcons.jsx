import React from 'react';

const navigationIcons = {
  hamburger: (
    <svg width={44} height={44}>
      <path
        d="M12 29h21v-2.333H12V29zm0-5.833h21v-2.334H12v2.334zM12 15v2.333h21V15H12z"
        fill="#FFF"
      />
    </svg>
  ),
  cross: (
    <svg width={44} height={44}>
      <path
        fill="#FFF"
        d="M26.741 15L21.6 20.142 16.458 15 15 16.458l5.142 5.142L15 26.742l1.458 1.458 5.142-5.141 5.141 5.141 1.459-1.458-5.142-5.142 5.142-5.142z"
        fillRule="evenodd"
      />
    </svg>
  ),
};

export default navigationIcons;
