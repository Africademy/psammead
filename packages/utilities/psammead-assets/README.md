# psammead-assets - [![Known Vulnerabilities](https://snyk.io/test/github/bbc/psammead/badge.svg?targetFile=packages%2Futilities%2Fpsammead-assets%2Fpackage.json)](https://snyk.io/test/github/bbc/psammead?targetFile=packages%2Futilities%2Fpsammead-assets%2Fpackage.json) [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-assets.svg)](https://www.npmjs.com/package/@bbc/psammead-assets) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

This package provides a collection of common assets that are likely to be required by many Psammead components or users, such as SVGs or small scripts.

## Exports

`/amp-boilerplate` - A helper, allowing projects using Psammead to easily pull in a versioned copy of [AMP's required boilerplate scripts](https://github.com/ampproject/amphtml/blob/master/spec/amp-boilerplate.md).
`/svgs` - SVG icons commonly required by projects using Psammead.

## Installation

```jsx
npm install @bbc/psammead-assets --save
```
## Usage

```jsx
import { BBC_BLOCKS, news, getSVG } from '@bbc/psammead-assets/svgs';

import { AMP_SCRIPT } from '@bbc/psammead-assets/amp-boilerplate';

```
## getSVG

`getSVG` returns an svg with the correct scaling. It is intended to be used in storybooks.

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| group | node | no | |```<g fillrule="evenodd"><path d="M84.32,0V24H58.82V0ZM79.26,16l-.41.25a11.91,11.91,0,0,1-6.07,1.85c-4.18,0-6.93-2.49-6.94-6.09s2.88-6.13,6.83-6.14a12,12,0,0,1,6,1.71l.4.22V4.6l-.17-.07A16.54,16.54,0,0,0,72.7,3.14a10.56,10.56,0,0,0-7.19,2.57,8.68,8.68,0,0,0-2.85,6.54A8.46,8.46,0,0,0,65,17.93a10,10,0,0,0,7.5,2.94h0a14,14,0,0,0,6.56-1.5l.15-.07Z M54.94,0V24H29.43V0ZM48.88,15.91a4.62,4.62,0,0,0-3.77-4.47,4.25,4.25,0,0,0,1.49-1.19,3.75,3.75,0,0,0,.72-2.34,4.17,4.17,0,0,0-1.39-3.13,6.33,6.33,0,0,0-4.36-1.41H36.12V20.65H42.5a7.18,7.18,0,0,0,4.9-1.53A4.27,4.27,0,0,0,48.88,15.91Z M45.7,15.57A2,2,0,0,1,45,17.16a4.22,4.22,0,0,1-2.85.79H39.22V13.23H42a4.89,4.89,0,0,1,2.82.68A1.93,1.93,0,0,1,45.7,15.57Z M43.32,9.92a2,2,0,0,0,.82-1.79,1.82,1.82,0,0,0-.58-1.43,3.33,3.33,0,0,0-2.26-.63H39.22v4.47h1.43A4.62,4.62,0,0,0,43.32,9.92Z M25.55,0V24H0V0ZM19.49,15.91a4.62,4.62,0,0,0-3.77-4.47,4.24,4.24,0,0,0,1.49-1.19,3.75,3.75,0,0,0,.72-2.34,4.18,4.18,0,0,0-1.39-3.13,6.33,6.33,0,0,0-4.36-1.41H6.74V20.65h6.38A7.18,7.18,0,0,0,18,19.12,4.27,4.27,0,0,0,19.49,15.91Z M13.93,9.92a2,2,0,0,0,.82-1.79,1.82,1.82,0,0,0-.58-1.43,3.33,3.33,0,0,0-2.26-.63H9.83v4.47h1.43A4.62,4.62,0,0,0,13.93,9.92Z M15.47,13.91a4.89,4.89,0,0,0-2.82-.68H9.83V18h2.94a4.23,4.23,0,0,0,2.85-.79,2,2,0,0,0,.69-1.59A1.93,1.93,0,0,0,15.47,13.91Z" /><path d="M111,3.36h2.36V20.7h-2.13L99.66,7.35V20.7H97.33V3.36h2L111,16.83Z" /><path d="m117.86 3.36h9.83v2.21h-7.35v5.29h7.1v2.22h-7.1v5.39h7.58v2.21h-10.06z M154.18,3.36h2.48l-7,17.41h-.55l-5.67-14.1-5.73,14.1h-.53l-7-17.41h2.5l4.78,12,4.81-12h2.35l4.83,12Z M163.38,13.43l-1.89-1.15A8.57,8.57,0,0,1,159,10.16a4,4,0,0,1-.75-2.41,4.26,4.26,0,0,1,1.42-3.33,5.31,5.31,0,0,1,3.69-1.28,7,7,0,0,1,4,1.22V7.17a5.74,5.74,0,0,0-4-1.8,3.34,3.34,0,0,0-2,.56,1.71,1.71,0,0,0-.78,1.44,2.22,2.22,0,0,0,.58,1.46,7.25,7.25,0,0,0,1.85,1.43l1.9,1.12Q168,13.28,168,16.21a4.42,4.42,0,0,1-1.4,3.39A5.11,5.11,0,0,1,163,20.9a7.63,7.63,0,0,1-4.68-1.58V16.17a5.84,5.84,0,0,0,4.65,2.55,2.93,2.93,0,0,0,1.94-.65,2,2,0,0,0,.78-1.63Q165.67,14.85,163.38,13.43Z" /> </g>```s|
| ratio | number | no | 6.9979 | N/A |
| height | number | no | 24 | N/A |

## getSVG Usage

```jsx
import { getSVG, news } from '@bbc/psammead-assets';

const Svg = getSVG({ group: news.group, ratio: news.ratio, height: 24 });

const WrapperComponent = () => (
  <div>
    <Svg/>
  </div>
);
```


## Contributing

When **adding** a new export to this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/psammead-assets/index.test.jsx#L11-L18) also need to be updated. When **removing** an exisiting export from this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/psammead-assets/index.test.jsx#L11-L18) need to be updated and the package version requires a major change (EG: 1.2.1 -> 2.0.0) as this would be considered a breaking change due to functionality being removed.

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
