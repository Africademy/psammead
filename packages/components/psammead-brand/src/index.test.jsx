import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import * as svgs from '@bbc/psammead-assets/svgs';
import Brand from './index';

describe('Brand', () => {
  shouldMatchSnapshot(
    'should render correctly with News svg',
    <Brand brandName="Default Brand Name" svg={svgs.news} />,
  );
  shouldMatchSnapshot(
    'should render correctly with Igbo service svg',
    <Brand brandName="Default Brand Name" svg={svgs.igbo} />,
  );
  shouldMatchSnapshot(
    'should render correctly with Yoruba service svg',
    <Brand brandName="Default Brand Name" svg={svgs.yoruba} />,
  );
  shouldMatchSnapshot(
    'should render correctly with Thai service svg',
    <Brand brandName="Default Brand Name" svg={svgs.thai} />,
  );
  shouldMatchSnapshot(
    'should render correctly with Pidgin service svg',
    <Brand brandName="Default Brand Name" svg={svgs.pidgin} />,
  );
});
