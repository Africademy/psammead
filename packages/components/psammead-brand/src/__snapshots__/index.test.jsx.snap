// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand should render correctly with link not provided 1`] = `
.c2 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
}

.c1 {
  box-sizing: content-box;
  fill: #FFFFFF;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c3:hover .c1,
.c3:focus .c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:37.5rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c1 {
    fill: windowText;
  }
}

<div
  className="c0"
>
  <svg
    aria-hidden="true"
    className="c1"
    focusable="false"
    height={24}
    viewBox="0 0 167.95 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fillrule="evenodd"
    >
      <path
        d="M84.32"
      />
    </g>
  </svg>
  <span
    className="c2"
  >
    <span
      lang="en-GB"
    >
      Default Brand Name
    </span>
    Service
  </span>
</div>
`;

exports[`Brand should render correctly with link provided 1`] = `
.c4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
}

.c2 {
  display: inline-block;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c3 {
  box-sizing: content-box;
  fill: #FFFFFF;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c1:hover .c3,
.c1:focus .c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:37.5rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c3 {
    fill: windowText;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    href="https://www.bbc.co.uk/news"
  >
    <svg
      aria-hidden="true"
      className="c3"
      focusable="false"
      height={24}
      viewBox="0 0 167.95 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillrule="evenodd"
      >
        <path
          d="M84.32"
        />
      </g>
    </svg>
    <span
      className="c4"
    >
      <span
        lang="en-GB"
      >
        Default Brand Name
      </span>
      Service
    </span>
  </a>
</div>
`;

exports[`Brand should render correctly with no service Localized Name 1`] = `
.c2 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

.c0 {
  background-color: #B80000;
  height: 3.5rem;
  width: 100%;
  padding: 0 0.5rem;
}

.c1 {
  box-sizing: content-box;
  fill: #FFFFFF;
  padding-top: 1rem;
  padding-bottom: 0.75rem;
  width: 100%;
  max-width: 17.5rem;
  min-width: 11.25rem;
}

.c3:hover .c1,
.c3:focus .c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

@media (min-width:37.5rem) {
  .c0 {
    height: 5rem;
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding-top: 1.75rem;
    padding-bottom: 1.5rem;
  }
}

@media screen and (-ms-high-contrast:active),print {
  .c1 {
    fill: windowText;
  }
}

<div
  className="c0"
>
  <svg
    aria-hidden="true"
    className="c1"
    focusable="false"
    height={24}
    viewBox="0 0 167.95 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fillrule="evenodd"
    >
      <path
        d="M84.32"
      />
    </g>
  </svg>
  <span
    className="c2"
  >
    BBC News
  </span>
</div>
`;
